From 864b0a53782bae1299b77f42efa6677c6389e533 Mon Sep 17 00:00:00 2001
From: Felix Singer <felixsinger@posteo.net>
Date: Sat, 3 Jun 2023 00:07:01 +0200
Subject: [PATCH] Gemfile: Fix dependenies

For NixOS, just load the PostgreSQL and the MySQL gem.

To fix the ffi gem, add `:mri` to platforms.

Since webrick is not included anymore, add it.

Signed-off-by: Felix Singer <felixsinger@posteo.net>
---
 Gemfile | 37 +++++--------------------------------
 1 file changed, 5 insertions(+), 32 deletions(-)

diff --git a/Gemfile b/Gemfile
index 3f2aa5ab8..fb5165498 100644
--- a/Gemfile
+++ b/Gemfile
@@ -54,37 +54,8 @@ group :common_mark do
   gem 'deckar01-task_list', '2.3.2'
 end
 
-# Include database gems for the adapters found in the database
-# configuration file
-require 'erb'
-require 'yaml'
-database_file = File.join(File.dirname(__FILE__), "config/database.yml")
-if File.exist?(database_file)
-  yaml_config = ERB.new(IO.read(database_file)).result
-  database_config = YAML.respond_to?(:unsafe_load) ? YAML.unsafe_load(yaml_config) : YAML.load(yaml_config)
-  adapters = database_config.values.map {|c| c['adapter']}.compact.uniq
-  if adapters.any?
-    adapters.each do |adapter|
-      case adapter
-      when 'mysql2'
-        gem "mysql2", "~> 0.5.0", :platforms => [:mri, :mingw, :x64_mingw]
-      when /postgresql/
-        gem "pg", "~> 1.2.2", :platforms => [:mri, :mingw, :x64_mingw]
-      when /sqlite3/
-        gem "sqlite3", "~> 1.4.0", :platforms => [:mri, :mingw, :x64_mingw]
-      when /sqlserver/
-        gem "tiny_tds", "~> 2.1.2", :platforms => [:mri, :mingw, :x64_mingw]
-        gem "activerecord-sqlserver-adapter", "~> 6.1.0", :platforms => [:mri, :mingw, :x64_mingw]
-      else
-        warn("Unknown database adapter `#{adapter}` found in config/database.yml, use Gemfile.local to load your own database gems")
-      end
-    end
-  else
-    warn("No adapter found in config/database.yml, please configure it first")
-  end
-else
-  warn("Please configure your config/database.yml first")
-end
+gem "mysql2", "~> 0.5.0", :platforms => [:mri, :mingw, :x64_mingw]
+gem "pg", "~> 1.2.2", :platforms => [:mri, :mingw, :x64_mingw]
 
 group :development do
   gem 'listen', '~> 3.3'
@@ -95,7 +66,7 @@ group :test do
   gem "rails-dom-testing"
   gem 'mocha', (Gem.ruby_version < Gem::Version.new('2.7.0') ? ['>= 1.4.0', '< 2.0.0'] : '>= 1.4.0')
   gem 'simplecov', '~> 0.21.2', :require => false
-  gem "ffi", platforms: [:mingw, :x64_mingw, :mswin]
+  gem "ffi", platforms: [:mri, :mingw, :x64_mingw, :mswin]
   # For running system tests
   gem 'puma', (Gem.ruby_version < Gem::Version.new('2.7') ? '< 6.0.0' : '>= 0')
   gem 'capybara', (if Gem.ruby_version < Gem::Version.new('2.6')
@@ -122,3 +93,5 @@ end
 Dir.glob File.expand_path("../plugins/*/{Gemfile,PluginGemfile}", __FILE__) do |file|
   eval_gemfile file
 end
+
+gem 'webrick'
-- 
2.38.5

